File: children/src/main/java/io/innofang/children/reminder/ReminderViewPagerAdapter.java
package io.innofang.children.reminder;

import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;

import java.util.ArrayList;
import java.util.List;

/**
 * Author: Inno Fang
 * Time: 2017/12/2 11:13
 * Description:
 */


public class ReminderViewPagerAdapter extends FragmentPagerAdapter {

    private List<Fragment> mFragmentList = new ArrayList<>();
    private String[] titles;

    public ReminderViewPagerAdapter(FragmentManager fm, String[] titles) {
        super(fm);
        this.titles = titles;
    }

    public void addFragment(Fragment fragment) {
        mFragmentList.add(fragment);
    }

    @Override
    public Fragment getItem(int position) {
        return mFragmentList.get(position);
    }

    @Override
    public int getCount() {
        return mFragmentList.size();
    }

    @Override
    public CharSequence getPageTitle(int position) {
        return titles[position];
    }
}

File: children/src/main/java/io/innofang/children/reminder/ReminderViewPagerAdapter.kt

package cn.edu.nuc.studies

import android.support.v4.app.Fragment
import android.support.v4.app.FragmentManager
import android.support.v4.app.FragmentPagerAdapter



/**
 * Author: Inno Fang
 * Time: 2017/8/15 22:58
 * Description:
 */


class ReminderViewPagerAdapter
constructor(fm: FragmentManager, private val titles: Array<String>) : FragmentPagerAdapter(fm) {

    private val mFragmentList =  java.util.ArrayList<Fragment>()

    fun addFragment(fragment: Fragment) {
        mFragmentList.add(fragment)
    }

    override fun getItem(position: Int): Fragment {
        return mFragmentList[position]
    }

    override fun getCount(): Int {
        return mFragmentList.size
    }

    override fun getPageTitle(position: Int): CharSequence {
        return titles[position]
    }
}